account_credit_control/migrations/13.0.2.0.0/post-migration.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2020 Tecnativa - Jairo Llopis
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+from openupgradelib import openupgrade
+
+from odoo.tools.parse_version import parse_version
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    # Skip migration if 12.0.3 was already installed (migration was already run)
+    if parse_version("12.0.3.0.0") <= parse_version(version) < parse_version("13.0"):
```

Comment:
I'm not sure this will work in any case. Why don't you simply check if a column, a table or similar exists for deducing if the migration script has already been executed in 12.0?

Reply:
AFAICS it's gonna be the same, and this is simpler and more self-explaining IMHO. :shrug: 

Check out this shell session:

```py
>>> from odoo.tools.parse_version import parse_version                                                                                                                                                                

>>> version = "12.0.2.9999.9999"                                                                                                                                                                                      

>>> parse_version("12.0.3.0.0") <= parse_version(version) < parse_version("13.0")                                                                                                                                     
False


>>> version = "12.0.3.0.0"                                                                                                                                                                                            

>>> parse_version("12.0.3.0.0") <= parse_version(version) < parse_version("13.0")                                                                                                                                     
True


>>> version = "12.0.4.0.0"                                                                                                                                                                                            

>>> parse_version("12.0.3.0.0") <= parse_version(version) < parse_version("13.0")                                                                                                                                     
True


>>> version = "13.0.0.0.0"                                                                                                                                                                                            

>>> parse_version("12.0.3.0.0") <= parse_version(version) < parse_version("13.0")                                                                                                                                     
False


>>> version = "13.0.1.0.0"                                                                                                                                                                                            

>>> parse_version("12.0.3.0.0") <= parse_version(version) < parse_version("13.0")                                                                                                                                     
False
```

Reply:
I don't share your clarity reading that code, but we will see if we find problems. Anyway, I have been checking the scripts, and it seems they can be re-entrant.